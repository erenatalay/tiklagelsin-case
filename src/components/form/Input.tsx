import React, { FC } from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { TextInput, TextInputProps } from 'react-native-paper';
import ErrorText from './ErrorText';
interface InputProps extends TextInputProps {
    name: string
    touched?: boolean
    changeText: (input: string, text: string) => void
    blur: (input: string) => void
    errors?: string
}
const Input: FC<InputProps> = (props) => {
    const { name, placeholder, left, keyboardType, value, theme, changeText, blur, touched, secureTextEntry = false, errors } = props;
    return (
        <>
            <TextInput
                theme={theme}
                left={left}
                value={value}
                onBlur={() => blur(name)}
                keyboardType={keyboardType}
                placeholder={placeholder}
                // placeholderTextColor={COLORS.textInput}
                // activeOutlineColor={COLORS.textInput}
                secureTextEntry={secureTextEntry}
                autoComplete='off'
                style={[styles.textInput, { marginBottom: 15,height : props.multiline ? 60 : 30 }]}
                mode={"outlined"}
                onChangeText={(text: string) => changeText(name, text)}
                {...props}
            />
            <ErrorText error={errors} touched={touched} />
        </>
    )
}
const styles = StyleSheet.create({
    textInput: {
        paddingVertical: 15,
        borderRadius: 64,
        height:  30,
        justifyContent : "center",
    },
})
export default Input;