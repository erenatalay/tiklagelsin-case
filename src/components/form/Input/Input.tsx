import React, { FC } from 'react'
import { StyleSheet, View, useColorScheme } from 'react-native'
import { TextInput, TextInputProps, DefaultTheme } from 'react-native-paper';
import ErrorText from '../ErrorText/ErrorText';
import Colors from '../../../constant/Colors';
interface InputProps extends TextInputProps {
    name: string
    touched?: boolean
    changeText: (input: string, text: string) => void
    blur: (input: string) => void
    errors?: string
}
const Input: FC<InputProps> = (props) => {
    const { name, placeholder, left, keyboardType, value, changeText, blur, touched, secureTextEntry = false, errors } = props;
    const color = useColorScheme() ?? 'light';
    const theme = {
        ...DefaultTheme,
        colors: {
            ...DefaultTheme.colors,
            background: "white",
            text: "blue",
        },
        ...{ roundness: 30 },

    };
    return (
        <View style={styles.container}>
            <TextInput
                testID='text-input'
                theme={theme}
                left={left}
                value={value}
                onBlur={() => blur(name)}
                keyboardType={keyboardType}
                placeholder={placeholder}
                placeholderTextColor={Colors[color].textInputPlaceholder}
                activeOutlineColor={Colors[color].textInputBackground}
                secureTextEntry={secureTextEntry}
                autoComplete='off'
                style={[styles.textInput, { height: props.multiline ? 60 : 30 }]}
                mode={"outlined"}
                onChangeText={(text: string) => changeText(name, text)}
                {...props}
            />
            <ErrorText error={errors} touched={touched} />
        </View>
    )
}
const styles = StyleSheet.create({
    container :{
        marginVertical : 10
    },
    textInput: {
        paddingVertical: 15,
        borderRadius: 64,
        height: 30,
        justifyContent: "center",
    },
})
export default Input;