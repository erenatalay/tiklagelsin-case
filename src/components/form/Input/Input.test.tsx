import React from 'react';
import Input from './Input';
import { render, fireEvent, screen } from '@testing-library/react-native';
describe('Input component tests', () => {
    it('is rendered correctly', () => {
        render(
            <Input
            name="inputName"
            placeholder="Enter text"
            changeText={() => null}
            blur={() => null}

        />);
        const componentElement = screen.getByTestId('text-input');
        expect(componentElement).toBeDefined();
    });

    it('should trigger input when typed', () => {
        const mockSearch = jest.fn();
        const wrapper = render(
            <Input
                name="inputName"
                placeholder="Enter text"
                changeText={mockSearch}
                blur={() => null}

            />
        );
        const input = wrapper.getByTestId('text-input');
        fireEvent(input, 'onChangeText', 'test');
        expect(mockSearch).toHaveBeenCalled();
    });
});

