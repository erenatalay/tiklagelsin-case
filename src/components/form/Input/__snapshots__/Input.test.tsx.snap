// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests is rendered correctly 1`] = `
<Input
  blur={[Function]}
  changeText={[Function]}
  name="inputName"
  placeholder="Enter text"
/>
`;
